api/src/main/java/io/strimzi/api/kafka/model/KafkaAuthorizationKeycloak.java

```diff
@@ -0,0 +1,114 @@
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.strimzi.crdgenerator.annotations.Example;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.util.List;
+
+/**
+ * Configures Keycloak authorization on the brokers
+ */
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"type", "clientId", "tokenEndpointUri",
+                    "tlsTrustedCertificates", "disableTlsHostnameVerification",
+                    "delegateToKafkaAcls", "superUsers"})
+@EqualsAndHashCode
+public class KafkaAuthorizationKeycloak extends KafkaAuthorization {
+    private static final long serialVersionUID = 1L;
+
+    public static final String TYPE_KEYCLOAK = "keycloak";
+
+    public static final String AUTHORIZER_CLASS_NAME = "io.strimzi.kafka.oauth.server.authorizer.KeycloakRBACAuthorizer";
+    public static final String PRINCIPAL_BUILDER_CLASS_NAME = "io.strimzi.kafka.oauth.server.authorizer.JwtKafkaPrincipalBuilder";
+
+    private String clientId;
+    private String tokenEndpointUri;
+    private List<CertSecretSource> tlsTrustedCertificates;
+    private boolean disableTlsHostnameVerification = false;
+    private boolean delegateToKafkaAcls = false;
+    private List<String> superUsers;
+
+    @Description("Must be `" + TYPE_KEYCLOAK + "`")
+    @Override
+    public String getType() {
+        return TYPE_KEYCLOAK;
+    }
+
+    @Description("OAuth Client ID which the Kafka client can use to authenticate against the OAuth server and use the token endpoint URI.")
+    @JsonProperty(required = true)
+    public String getClientId() {
+        return clientId;
+    }
+
+    public void setClientId(String clientId) {
+        this.clientId = clientId;
+    }
+
+    @Description("Authorization server token endpoint URI.")
+    @JsonProperty(required = true)
+    public String getTokenEndpointUri() {
+        return tokenEndpointUri;
+    }
+
+    public void setTokenEndpointUri(String tokenEndpointUri) {
+        this.tokenEndpointUri = tokenEndpointUri;
+    }
+
+    @Description("Trusted certificates for TLS connection to the OAuth server.")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public List<CertSecretSource> getTlsTrustedCertificates() {
+        return tlsTrustedCertificates;
+    }
+
+    public void setTlsTrustedCertificates(List<CertSecretSource> tlsTrustedCertificates) {
+        this.tlsTrustedCertificates = tlsTrustedCertificates;
+    }
+
+    @Description("Enable or disable TLS hostname verification. " +
```

Comment:
hostname verification of _what_? Presumably the authz serverm but you should say that.

And should it be "enable...", defaulting to true?

Reply:
I used the same convention of the other OAuth authentication classes (i.e. see `KafkaListenerAuthenticationOAuth`)

Reply:
OK